### PR TITLE
LockVisualizer

### PR DESCRIPTION
**Pull request by MilenVolf**
## About the PR
Splits part of code in `EntityStorageVisualizerSystem` into its own `LockVisualizerSystem` with `LockVisualsComponent`.
Besides, fixes deployable barrier deployed state not being shown when locked.

## Why / Balance
Was required in TODO in `EntityStorageVisualizerSystem`.

Besides, I wanted to make GlassBox lock state work so I've faced with the problem of LockVisuals not being geneirc.

## Technical details
Methods in systems that update the lock state were cleaned up in favour of using a more generic LockVisualizer in prototype.
Basically, `LockVisualizerSystem` is just copy-pasted part of `EntityStorageVisualizerSystem` with some fixes.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

Content Client_2iVUY7ePiI

## Breaking changes

**Changelog**
N/A

### CODE REVIEW

**Comment by MilenVolf** [comment]
It's little messy in `LockVisualizerSystem` but any suggestion will be appreciated.

`LockVisualsComponent` besides `StorageUnlocked` has `StateStorageUnlocked` for storages that surely must have "unlocked" state.
But if you want to make an entity with unlocked state, you will have to put it manually in `LockVisuals` in prototype.

**Comment by MilenVolf** [comment]
Will try to clean that mess

**Review by mirrorcult — APPROVED** [review]
